src/rust/runtime/memory/buffer/databuffer.rs

```diff
@@ -73,10 +73,6 @@ impl DataBuffer {
 
     /// Creates a data buffer from a raw pointer and a length.
     pub fn from_raw_parts(data: *mut u8, len: usize) -> Result<Self, Fail> {
-        // Check if arguments are valid.
```

Comment:
Does catloop use zero-length buffers?

Reply:
No, this was a bug. But note that it does trick the sga length for encoding the magic close.

Reply:
I don't understand.  What purpose does moving the `len` check until after the `data` check serve?

But more importantly, we're trying to remove the DataBuffer type from the system, I only left it in place since Anand was in the middle of creating the Windows-compatible version of catnap.  So, the only changes we should be making to this file is to delete it.

Reply:
I unresolved this, as I would still like to know why this file is included in this PR.

Reply:
I reverted this file to its original state. Sorry for touching on it.